packages/medusa/src/models/line-item-adjustment.ts

```diff
@@ -7,13 +7,15 @@ import {
   JoinColumn,
   PrimaryColumn,
   OneToOne,
+  Unique
 } from "typeorm"
 import { ulid } from "ulid"
 import { DbAwareColumn } from "../utils/db-aware-column"
 import { Discount } from "./discount"
 import { LineItem } from "./line-item"
 
 @Entity()
+@Index(["discount_id", "item_id"], { unique: true, where: `"discount_id" IS NULL` })
```

Comment:
Should this not be `where: "discount_id" IS NOT NULL`?

Reply:
nit: For other unique constraints, we've used the `@Unique` decorator like so:
```
@Unique("uniqitemdisc", ["discount_ite", "item_id"])
```

Reply:
Would be great to have a test that validates the uniqueness constraint.

Reply:
> Should this not be `where: "discount_id" IS NOT NULL`?

Yes definitely! Sorry, I had it on the `feat/line-item-adjustments-draft-orders` PR [here](https://github.com/medusajs/medusa/blob/be46e5cb14b88e23333cfa3c482e6845ac324d7f/packages/medusa/src/models/line-item-adjustment.ts#L18). Will fix in this branch too!

> nit: For other unique constraints, we've used the `@Unique` decorator like so:
> 
> ```
> @Unique("uniqitemdisc", ["discount_ite", "item_id"])
> ```

Will it enforce the constraint only if the `discount_id` is not null?  I tried using that at first, but couldn't find a way to tell typeorm to only enforce the unique constraint when `discount_id` is not null. Did you achieve something similar in the past using `@Unique()` ?



> Would be great to have a test that validates the uniqueness constraint.

Very true! Will add a few 👍